### ISSUE TITLE
syntax: Needless braces aren't removed.

### ISSUE THREAD

**Issue by RalphCorderoy** [mvdan/sh#99]
Needless braces around variable names, favoured by some unfamiliar with sh programming, are clutterful noise from the land of HTML templates that need removing, just as `go fmt` turns `(((42)))` into `(42)`.

    echo "You are ${USER} at ${HOME}."
    echo "You are $USER at $HOME."

**Comment by mvdan** [comment]
See my reply to #100. In particular, I thought of this very "simplification" in #34 and decided against it.

**Comment by mvdan** [comment]
ping @RalphCorderoy

**Comment by mvdan** [comment]
Since there are no further arguments, I'm going to close this since I already decided against this in #34.

@RalphCorderoy if you still think it's a good idea, feel free to reopen - but read my reasoning in that issue first.

**Comment by mmlb** [comment]
o/ Hi @mvdan. Sorry for the necropost, but I prefer having discussions in minimal amount of issues if possible to keep search results from growing unnecessarily.

I've just come here after seeing a large-ish code base that has unnecessary use of `${var}` (along with `var="unnecessaryquotes"`). I'd love to see both removed via `-s`. We had a little bit of a discussion of this in #117, and I'm still here wishing we could have this. Both shfmt and shellcheck are tools I've come to trust quite a bit, even with their flaws. shellcheck sometimes has me quote things that if it was smarter it would realize were unnecessary but the benefit of shellcheck outweighs a couple of extra quotes here and there. Similarly shfmt stripping unnecessary code has led me to go searching for the reasoning and improved my bash knowledge. shfmt changes has also led to discussions where multiple people have learned new things.

I'm not sure I've ever come across a code base that has fully consistent use of `${}` afaicr. On the other hand its just too hard sometimes to go from inconsistent to consistent avoidance of unnecessary `${}` (which I think is valuable for the reason I stated in previous discussion: avoiding unnecessary `${}` helps learn what the {} is really for and also avoids hard to reach chars on most keyboards :smile:), especially when there's no tool make it easy/automated. We'll just waste time going from consistent -> inconsistent and fixing, exactly what *fmt tools are good at avoiding.

Anyways, just wanted to provide a data point that many years later I've found myself fighting this issue and looking to shfmt for some help. I'd love for shfmt to form an opinion and enforce it, even if it was `${}` just so we could get to more consistent code bases.
